packages/agreement-process/test/activateAgreement.test.ts

```diff
@@ -0,0 +1,2171 @@
+        ...getMockEService(),
+        producerId: producer.id,
+        descriptors: [descriptor],
+      };
+
+      const agreement: Agreement = {
+        ...getMockAgreement(),
+        state: agreementState.pending,
+        eserviceId: eservice.id,
+        descriptorId: descriptor.id,
+        producerId: producer.id,
+        consumerId: consumer.id,
+      };
+
+      await addOneTenant(consumer);
+      await addOneTenant(producer);
+      await addOneEService(eservice);
+      await addOneAgreement(agreement);
+
+      await expect(
+        agreementService.activateAgreement(agreement.id, {
+          authData,
+          serviceName: "",
+          correlationId: "",
+          logger: genericLogger,
+        })
+      ).rejects.toThrowError(agreementActivationFailed(agreement.id));
+    });
+
+    it("Agreement Pending, Requester === Consumer -- error case: throws operationNotAllowed", async () => {
+      const consumerId = generateId<TenantId>();
+      const authData = getRandomAuthData(consumerId);
+
+      const agreement: Agreement = {
+        ...getMockAgreement(),
+        state: agreementState.pending,
+        consumerId,
+      };
+      await addOneAgreement(agreement);
+      await expect(
+        agreementService.activateAgreement(agreement.id, {
+          authData,
+          serviceName: "",
+          correlationId: "",
+          logger: genericLogger,
+        })
+      ).rejects.toThrowError(operationNotAllowed(authData.organizationId));
+    });
+  });
+
+  describe("Agreement Suspended", () => {
+    it.each([
+      "Requester === Producer, suspendedByProducer === true, suspendedByConsumer === false",
+      "Requester === Consumer, suspendedByProducer === false, suspendedByConsumer === true",
+    ])(
+      "Agreement Suspended, valid attributes, %s -- success case: Suspended >> Activated",
+      async (s) => {
+        const producer: Tenant = getMockTenant();
+
+        const validTenantCertifiedAttribute: CertifiedTenantAttribute = {
+          ...getMockCertifiedTenantAttribute(),
+          revocationTimestamp: undefined,
+        };
+
+        const validTenantDeclaredAttribute: DeclaredTenantAttribute = {
+          ...getMockDeclaredTenantAttribute(),
+          revocationTimestamp: undefined,
+        };
+
+        const validTenantVerifiedAttribute: VerifiedTenantAttribute = {
+          ...getMockVerifiedTenantAttribute(),
+          verifiedBy: [
+            {
+              id: producer.id,
+              verificationDate: new Date(),
+              extensionDate: new Date(new Date().getTime() + 3600 * 1000),
+            },
+          ],
+        };
+
+        const consumer: Tenant = {
+          ...getMockTenant(),
+          attributes: [
+            validTenantCertifiedAttribute,
+            validTenantDeclaredAttribute,
+            validTenantVerifiedAttribute,
+          ],
+        };
+
+        const isProducer = s.startsWith("Requester === Producer");
```

Comment:
this is a bit difficult to maintain, and values for `suspendedByProducer` and `suspendedByConsumer`  are based on assumptions of the description as written today.
Is it possible to pass an object in the `it.each` with the actual values we want to use?
Or in alternative, we could use a loop